src/sources/mod.rs

```diff
@@ -247,11 +268,481 @@ pub enum Sources {
+            Self::InternalLogs(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-internal_metrics")]
+            Self::InternalMetrics(config) => config.can_acknowledge(),
+            #[cfg(all(unix, feature = "sources-journald"))]
+            Self::Journald(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-kafka")]
+            Self::Kafka(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-kubernetes_logs")]
+            Self::KubernetesLogs(config) => config.can_acknowledge(),
+            #[cfg(all(feature = "sources-logstash"))]
+            Self::Logstash(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-mongodb_metrics")]
+            Self::MongodbMetrics(config) => config.can_acknowledge(),
+            #[cfg(all(feature = "sources-nats"))]
+            Self::Nats(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-nginx_metrics")]
+            Self::NginxMetrics(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-opentelemetry")]
+            Self::Opentelemetry(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-postgresql_metrics")]
+            Self::PostgresqlMetrics(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-prometheus")]
+            Self::PrometheusScrape(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-prometheus")]
+            Self::PrometheusRemoteWrite(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-redis")]
+            Self::Redis(config) => config.can_acknowledge(),
+            #[cfg(test)]
+            Self::TestBackpressure(config) => config.can_acknowledge(),
+            #[cfg(test)]
+            Self::TestBasic(config) => config.can_acknowledge(),
+            #[cfg(test)]
+            Self::TestError(config) => config.can_acknowledge(),
+            #[cfg(test)]
+            Self::TestPanic(config) => config.can_acknowledge(),
+            #[cfg(test)]
+            Self::TestTripwire(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-socket")]
+            Self::Socket(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-splunk_hec")]
+            Self::SplunkHec(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-statsd")]
+            Self::Statsd(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-stdin")]
+            Self::Stdin(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-syslog")]
+            Self::Syslog(config) => config.can_acknowledge(),
+            Self::UnitTest(config) => config.can_acknowledge(),
+            #[cfg(feature = "sources-vector")]
+            Self::Vector(config) => config.can_acknowledge(),
+        }
+    }
+}
+
+impl NamedComponent for Sources {
+    const NAME: &'static str = "_invalid_usage";
```

Comment:
Might be because of the long day, but where/how is this getting used?

Reply:
_This_ one will never get used because we'll never intentionally generate code that wants the associated constant version of the component name of `Sources`. We'll only ever get `get_component_name` on `Sources` which will get us the component name of whatever variant it is.

The future hope is eventually have the macros generate this on each variant in the big enums:

```rust
// In the big enum:
#[serde(rename_with = "<StdinConfig as NamedComponent>::NAME")]
Stdin(#[configurable(derived)] StdinConfig),
```